devenv/setup.sh

```diff
@@ -70,7 +70,8 @@ esac
 apt-get install -y linux-image-extra-$(uname -r) apparmor docker-engine
 
 # Configure docker
-echo "DOCKER_OPTS=\"-s=${DOCKER_STORAGE_BACKEND_STRING} -r=true --api-cors-header='*' -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock ${DOCKER_OPTS}\"" > /etc/default/docker
+DOCKER_OPTS="-s=${DOCKER_STORAGE_BACKEND_STRING} -r=true --api-cors-header='*' -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock ${DOCKER_OPTS}"
+sed -i.bak '/^DOCKER_OPTS=/{h;s|=.*|=\"'"${DOCKER_OPTS}"'\"|};${x;/^$/{s||DOCKER_OPTS=\"'"${DOCKER_OPTS}"'\"|;H};x}' /etc/default/docker
```

Comment:
If there is a line "DOCKER_OPTS=" in /etc/default/docker, then replace its value with ${DOCKER_OPTS}. Otherwise, add line DOCKER_OPTS=${DOCKER_OPTS} into /etc/default/docker.

Perhaps, a cleaner way exists to accomplish the same task. My main intention is to easily run vagrant behind proxy. Unfortunately, devenv does not work out of the box if you forced to use proxy.

See [this](http://superuser.com/questions/590630/sed-how-to-replace-line-if-found-or-append-to-end-of-file-if-not-found) answer on SO for more information.